### PR TITLE
move code in make_ped.nf to raredisease.nf

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

Pipeline creates a pedfile based on the information provided in the samplesheet, and this is done using a process (MAKE_PED). I have moved that functionality to the main workflow in this PR. This is done so that the file can be created in the nextflow headnode instead of starting a separate process for this small task. 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR> -stub`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test_one_sample,docker --outdir <OUTDIR> -stub`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/raredisease.nf [hunk 1]
```diff
@@ -560,6 +558,31 @@ workflow.onComplete {
...
+
+def make_ped(samples) {
+
+    def case_name  = samples[0].case_id
+    def outfile  = workDir.resolve("$case_name" + '.ped')
+    outfile.text = ['#family_id', 'sample_id', 'father', 'mother', 'sex', 'phenotype'].join('\t')
+    def samples_list = []
+    for(int i = 0; i<samples.size(); i++) {
+        sample_tokenized   =  samples[i].id.tokenize("_")
+        sample_tokenized.removeLast()
```
Comment (jemten) [inline workflows/raredisease.nf:575]: Do we always have a `_T` file ending even in hte case of a single sample-lane input?

File: workflows/raredisease.nf [hunk 2]
```diff
@@ -189,8 +188,7 @@ workflow RAREDISEASE {
                                                                               : Channel.value([])
 
     // Generate pedigree file
-    MAKE_PED (CHECK_INPUT.out.samples.toList())
-    ch_versions = ch_versions.mix(MAKE_PED.out.versions)
+    pedfile = CHECK_INPUT.out.samples.toList().map { make_ped(it) }
```
Comment (jemten) [inline workflows/raredisease.nf:191]: what does the `map` operator do here?

**Review by jemten — COMMENTED** [review]
Looks good! Have some comments but that's mainly down to my lack of nextflow/groovy knowledge 😅

File: workflows/raredisease.nf [hunk 3]
```diff
@@ -560,6 +558,31 @@ workflow.onComplete {
     }
 }
 
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    FUNCTIONS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+def make_ped(samples) {
```
[the remark this answers is not in the document]
Comment (ramprasadn) [reply workflows/raredisease.nf:567]: Yupe! You are right :D

File: workflows/raredisease.nf (hunk 2 above)
Comment (ramprasadn) [reply workflows/raredisease.nf:191]: It opens the channel and passes the contents of the channel to the function.

**Review by jemten — APPROVED** [review]
👍

**Pull request merged**